### ISSUE TITLE
`Path` tests fail on Windows

### ISSUE THREAD

**Issue by scordio** [assertj/assertj-core#2312]
Some tests introduced with #2292 are failing only on Windows. All of them are negative tests, so there seems to be no impact on the happy path.

The tests are currently disabled with:
```java
@DisabledOnOs(value = WINDOWS, disabledReason = "gh-FIXME")
```

**`should_fail_if_actual_is_not_readable`**

Currently, `actual.toFile().setReadable(false)` is used to remove the readable flag from `actual`. It seems like that this has no effect on Windows. We should drop the `File` API and use something from the NIO2 world to properly set this flag for both Unix and Windows file systems.

**`should_fail_if_actual_is_not_executable`**

Files on Windows seem to be considered as executable by default, and calling `actual.toFile().setExecutable(true)` has no effect. Similar to the `readable` case, we should drop the `File` API and use something from the NIO2 world to properly set this flag for both Unix and Windows file systems.

**`should_rethrow_IOException_as_UncheckedIOException`**

Most of them are related to `Files.newInputStream` calls, but there are also two related to `Files.newDirectoryStream` which also fail:

* `Paths_assertIsDirectoryContaining_with_Predicate_Test`
* `Paths_assertIsDirectoryNotContaining_with_Predicate_Test`

while the corresponding one with String parameter are not failing:

* `Paths_assertIsDirectoryContaining_with_String_Test`
* `Paths_assertIsDirectoryNotContaining_with_String_Test`

**Comment by scordio** [comment]
`Paths_assertIsDirectoryContaining_with_Predicate_Test` and `Paths_assertIsDirectoryNotContaining_with_Predicate_Test` are now passing after inverting the stubbing calls: 57680333d70827eba86c443a10ec3f840b088414

**Comment by joel-costigliola** [comment]
I wonder if inverting stubbing leading to a different behavior is something known in mockito, maybe we should reach out to them to see if it's normal?

**Comment by scordio** [comment]
I agree, I'll put together a small example and I'll reach out to them.

**Comment by scordio** [comment]
RE **`should_rethrow_IOException_as_UncheckedIOException`**

When the stubbing is done with:

```java
given(nioFilesWrapper.newInputStream(actual)).willThrow(cause);
```

the underlying `Files.newInputStream` is called because `nioFilesWrapper` is a spy (i.e., real methods are called by default) and `newInputStream` is not stubbed yet at that point.

On the other side, stubbing with:

```java
willThrow(cause).given(nioFilesWrapper).newInputStream(actual);
```

doesn't trigger the invocation of the underlying method because that method is not really invoked during stubbing.

I think the behavior makes sense considering our current use case. I don't see the need to raise an issue to Mockito.
